### PR TITLE
Add the use of frequency reserve on maxP reduction

### PR DESCRIPTION
**Pull request by FranckLecuyer**

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/entities/equipment/modification/GenerationDispatchEntity.java [hunk 1]
```diff
@@ -61,23 +68,54 @@ public void update(@NonNull ModificationInfos modificationInfos) {
...
     }
 
-    public static List<GeneratorsFilterEmbeddable> toEmbeddable(List<GeneratorsFilterInfos> generators) {
+    public static List<GeneratorsFilterEmbeddable> toEmbeddableFilterInfos(List<GeneratorsFilterInfos> generators) {
         return generators == null ? null : generators.stream()
             .map(generator -> new GeneratorsFilterEmbeddable(generator.getId(), generator.getName()))
             .collect(Collectors.toList());
     }
 
+    public static List<GeneratorsFrequencyReserveEntity> toEmbeddableFrequencyReserveInfos(List<GeneratorsFrequencyReserveInfos> generators) {
```
Comment (ayolab) [inline src/main/java/org/gridsuite/modification/server/entities/equipment/modification/GenerationDispatchEntity.java:87]: same here

**Review by ayolab — APPROVED** [review]
Tests OK
Code OK

File: src/main/java/org/gridsuite/modification/server/entities/equipment/modification/GenerationDispatchEntity.java (hunk 1 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/entities/equipment/modification/GenerationDispatchEntity.java]: Renamed

File: src/main/java/org/gridsuite/modification/server/entities/equipment/modification/GenerationDispatchEntity.java [hunk 2]
```diff
@@ -61,23 +68,54 @@ public void update(@NonNull ModificationInfos modificationInfos) {
...
+        if (generatorsFrequencyReserve == null) {
+            generatorsFrequencyReserve = toEmbeddableFrequencyReserveInfos(generationDispatchInfos.getGeneratorsFrequencyReserve());
+        } else {
+            generatorsFrequencyReserve.clear();
+            generatorsFrequencyReserve.addAll(toEmbeddableFrequencyReserveInfos(generationDispatchInfos.getGeneratorsFrequencyReserve()));
+        }
     }
 
-    public static List<GeneratorsFilterEmbeddable> toEmbeddable(List<GeneratorsFilterInfos> generators) {
+    public static List<GeneratorsFilterEmbeddable> toEmbeddableFilterInfos(List<GeneratorsFilterInfos> generators) {
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/entities/equipment/modification/GenerationDispatchEntity.java]: Renamed

**Pull request merged**